tests/osaka/eip7939_count_leading_zeros/test_count_leading_zeros.py

```diff
@@ -276,3 +278,95 @@ def test_clz_jump_operation(
         post[callee_address] = Account(storage={"0x00": expected_clz})
 
     state_test(pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.valid_from("Osaka")
+@pytest.mark.parametrize("bits", [0, 16, 64, 128, 255])
+def test_clz_initcode_context(state_test: StateTestFiller, pre: Alloc, bits: int):
+    """Test CLZ opcode behavior when creating a contract."""
+    code = Op.SSTORE(1, Op.CLZ(1 << bits)) + Op.STOP
+
+    code_address = pre.deploy_contract(code=code)
+
+    init_code = Op.SSTORE(0, Op.CLZ(1 << bits))
+
+    creation_code = (
+        init_code
+        + Op.EXTCODECOPY(
+            address=code_address,
+            dest_offset=0,
+            offset=0,
+            size=Op.EXTCODESIZE(code_address),
+        )
+        + Op.RETURN(0, Op.MSIZE)
+    )
+
+    sender_address = pre.fund_eoa()
+    contract_address = compute_create_address(address=sender_address, nonce=0)
+    tx = Transaction(
+        to=None,
+        gas_limit=6_000_000,
+        data=creation_code,
+        sender=sender_address,
+    )
+
+    expected_clz = 255 - bits
+
+    post = {
+        contract_address: Account(storage={"0x00": expected_clz}),
+    }
+
+    state_test(pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.valid_from("Osaka")
+@pytest.mark.parametrize("bits", [0, 1, 4, 8, 16])
+@pytest.mark.parametrize("opcode", [Op.CREATE, Op.CREATE2])
+def test_clz_initcode_create(state_test: StateTestFiller, pre: Alloc, bits: int, opcode: Op):
+    """Test CLZ opcode behavior when creating a contract."""
+    init_code = Op.SSTORE(0, Op.CLZ(1 << bits))
+
+    ext_code = Op.SSTORE(1, Op.CLZ(1 << bits))
+    ext_code_address = pre.deploy_contract(code=ext_code)
+
+    creation_code = init_code + Op.EXTCODECOPY(
```

Comment:
In this case we could pass `ext_code` as the data of the transaction (see line 362), and use `CALLDATACOPY` instead of `EXTCODECOPY` instead, just to save the extra contract deployment that the tests currently requires.

I think we can also remove `init_code` entirely.